### PR TITLE
Added support for VR headsets that do not provide stageParameters 

### PR DESCRIPTION
**Pull request by olga-microsoft**
**Description:**
Fixes issue of inconsistent head and controller height offset, for cases when stageParameters property on VR display is null.

**Changes Proposed:**
Provide default user height, if this is the case to be consistent with tracked controls.

### CODE REVIEW

File: src/components/look-controls.js [hunk 1]
```diff
@@ -17,7 +18,9 @@ module.exports.Component = registerComponent('look-controls', {
     enabled: {default: true},
     hmdEnabled: {default: true},
     reverseMouseDrag: {default: false},
-    standing: {default: true}
+    standing: {default: true},
+    userHeight: {default: 0},
```
Comment (dmarcos) [inline src/components/look-controls.js]: Are you using this property anywhere? it seems you are using userHeight from the camera component in `getUserHeight`

File: src/components/look-controls.js [hunk 2]
```diff
@@ -54,11 +57,21 @@ module.exports.Component = registerComponent('look-controls', {
...
     this.controls.update();
     this.updateOrientation();
     this.updatePosition();
   },
 
+  /**
+   * Return user height to use for standing poses, where a device doesn't provide an offset.
+   */
+  getUserHeight: function () {
+    var headEl = this.data.headElement || this.el.sceneEl.camera.el;
```
Comment (dmarcos) [inline src/components/look-controls.js]: What are the cases where the headElement is not the active camera?

File: src/components/look-controls.js [hunk 3]
```diff
@@ -17,7 +18,9 @@ module.exports.Component = registerComponent('look-controls', {
     enabled: {default: true},
     hmdEnabled: {default: true},
     reverseMouseDrag: {default: false},
-    standing: {default: true}
+    standing: {default: true},
+    userHeight: {default: 0},
+    headElement: {type: 'selector'}
```
Comment (dmarcos) [inline src/components/look-controls.js]: Can we add a description for the use cases of this property?

File: tests/components/look-controls.test.js [hunk 4]
```diff
@@ -52,4 +53,27 @@ suite('look-controls', function () {
       window.dispatchEvent(new Event('mouseup'));
     });
   });
+
+  suite('head-height', function () {
```
Comment (dmarcos) [inline tests/components/look-controls.test.js]: Thanks for the tests!

**Comment by olga-microsoft** [comment]
Thank you @dmarcos for review. I removed the properties under question. 
Merged master in order to get tests pass.
Please let me know, if there are any more concerns.

File: src/components/look-controls.js (hunk 2 above)
Comment (machenmusik) [reply src/components/look-controls.js]: (e.g. spectator camera?)

File: src/components/look-controls.js [hunk 5]
```diff
@@ -56,11 +57,21 @@ module.exports.Component = registerComponent('look-controls', {
...
     this.controls.update();
     this.updateOrientation();
     this.updatePosition();
   },
 
+  /**
+   * Return user height to use for standing poses, where a device doesn't provide an offset.
+   */
+  getUserHeight: function () {
+    var headEl = this.el.sceneEl.camera.el;
```
Comment (dmarcos) [inline src/components/look-controls.js]: A scene could have multiple cameras. In this case we probably only care about the camera on the entity where this component is setup. I would use the public API this way:
````javascript 
var el = this.el;
var userHeight = el.hasAttribute('camera') && el.getAttribute('camera').userHeight || DEFAULT_CAMERA_HEIGHT;
return userHeight;
````

Comment (dmarcos) [reply src/components/look-controls.js]: Does this work in your case?

Comment (dmarcos) [reply src/components/look-controls.js]: I'm worried about the coupling of look-controls and the camera right now. Moving userHeight to look-controls is something we have discussed in the past but we can do it in a separate PR

**Comment by dmarcos** [comment]
Thanks! Pretty close! A small couple of tweaks and we're ready to go.

File: src/components/look-controls.js (hunk 5 above)
Comment (olga-microsoft) [reply src/components/look-controls.js]: Yes, this works and looks better. Thank  you!

**Comment by dmarcos** [comment]
Nice! Thank you! Welcome as a contributor!

**Comment by olga-microsoft** [comment]
Thank you!

**Pull request merged**